ykrt/src/mt.rs

```diff
@@ -742,6 +765,123 @@ mod tests {
         }
     }
 
+    #[test]
+    fn locations_dont_get_stuck_tracing() {
+        const THRESHOLD: usize = 2;
+
+        let mut mtt = MTBuilder::new().hot_threshold(THRESHOLD).init();
+
+        const INC: u8 = 0;
+        const RESTART: u8 = 1;
+        let prog = Arc::new(vec![INC, INC, RESTART]);
+
+        struct IO {
+            prog: Arc<Vec<u8>>,
+            count: u64,
+            pc: usize,
+        }
+
+        #[interp_step]
```

Comment:
If this `interp_step` is going to be the same in lots of tests, shall we lift it out and re-use it?

Reply:
I agree, there's a growing amount of duplication, but I think that's best left to another PR. At the moment I'm mostly concentrating on making sure we test as much as possible, which is a big effort for my poor little brain!